### PR TITLE
Tricore

### PR DESCRIPTION
**Pull request by shahinsba**
Add lifter for tricore architecture.

### CODE REVIEW

**Comment by rhelmot** [comment]
amazing!!! please fix the following items and I will gladly merge this:

- linting is failing
- we're trying to move all new tests away from using nose, and towards a more standard unittest.TestCase class format. Can you migrate your tests to this spec?

File: tests/test_tricore.py [hunk 1]
```diff
@@ -1,19 +1,28 @@
 #!/usr/bin/env python3
+""" test_tricore.py
+A module for testing tricore lifter.
+"""
+# pylint: disable=wildcard-import
+# pylint: disable=unused-wildcard-import
```
Comment (twizmwazin) [inline tests/test_tricore.py:6]: Both of these should probably be on the specific line rather than module-level.

**Review by twizmwazin — COMMENTED** [review]
`super-with-arguments`, `consider-using-f-string`, `logging-format-interpolation`, `invalid-name`, `line-too-long` all seem like they should be fixed rather than ignored

**Comment by shahinsba** [comment]
@rhelmot thanks,
- the linting issues solved.
- I didn't use python unittest before, but migrating to it should not be a big problem. I will give it a try.

File: tests/test_tricore.py (hunk 1 above)
Comment (shahinsba) [reply tests/test_tricore.py]: Done. Also in other modules.

**Comment by shahinsba** [comment]
> `super-with-arguments`, `consider-using-f-string`, `logging-format-interpolation`, `invalid-name`, `line-too-long` all seem like they should be fixed rather than ignored

@twizmwazin Actually I'm not agree with all pylint warning messages.
**invalid-name:** is there, because pylint complains that even one word variables are not snake_case conform. For class names it complains that they are not PascalCase, but keeping their current names including underscore(_) improves readability.
**line-to-long:** In some cases breaking lines reduces readability and makes debugging more difficult.
**using-f-string:** Using f-strings needs to define more local variables that has much worse effect than using traditional string.format().
**logging-format-interpolation:** solved.
**super-with-arguments:** solved.

**Comment by rhelmot** [comment]
invalid-name and consider-using-f-string are both disabled by our linter configuration. Did you look at the [original failing lint run](https://dev.azure.com/angr/angr-platforms/_build/results?buildId=12638&view=logs&j=eb1ca712-fec5-5d93-deb0-40983fe61f2f&t=2942050b-d9d9-59ed-ae86-14d6c416ab46)? Are you using [our pylintrc](https://github.com/angr/angr-dev/blob/master/pylintrc) to do your local testing? I will have to disagree with you about line-too-long though. There are many ways to break a long line into multiple lines, and there's always a way to make code readable without breaking the width limit.

@twizmwazin I noticed that the pylintrc in angr-dev had diverged from the one in ci-settings. Can we do something to make sure that doesn't happen again?

**Comment by rhelmot** [comment]
Alright, one last question: what's the point of your loader, since all of your sample programs are ELFs? Can it be removed?

**Comment by shahinsba** [comment]
@rhelmot Some fields in tricore ELF format are different from its base standard. But for now it can be removed as these differences are not considered by this loader.
Can you remove it from the Pull Request? Or I should remove it from the branch?

**Comment by rhelmot** [comment]
In that case, please remove it from the branch.

**Comment by rhelmot** [comment]
Merged. Thank you so much!

**Pull request merged**